src/coreclr/src/gc/unix/cgroup.cpp

```diff
@@ -394,6 +480,110 @@ class CGroup
         return result;
     }
 
+    static bool GetCGroup1CpuLimit(uint32_t *val)
+    {
+        long long quota;
+        long long period;
+
+        quota = ReadCpuCGroupValue(CGROUP1_CFS_QUOTA_FILENAME);
+        if (quota <= 0)
+            return false;
+
+        period = ReadCpuCGroupValue(CGROUP1_CFS_PERIOD_FILENAME);
+        if (period <= 0)
+            return false;
+
+        ComputeCpuLimit(quota, period, val);
+
+        return true;
+    }
+
+    static bool GetCGroup2CpuLimit(uint32_t *val)
+    {
+        FILE *file = nullptr;
+        char *endptr = nullptr;
+        char *max_quota_string = nullptr;
+        char *period_string = nullptr;
+        char *space = nullptr;
+
+        const size_t lineLen = 1 /* space */ +
+                               (2 * 3 * sizeof(long long)) /* max space for 2 long longs */ +
+                               1 /* newline */ +
+                               1 /* null */;
+        char line[lineLen];
+        int sscanRet = 0;
+        long long quota = 0;
+        long long period = 0;
+
+        bool result = false;
+
+        if (s_cpu_cgroup_path == nullptr)
+            return false;
+
+        char filename[PATH_MAX];
+        snprintf(filename, PATH_MAX, "%s%s", s_cpu_cgroup_path, CGROUP2_CPU_MAX_FILENAME);
+        file = fopen(filename, "r");
+        if (file == nullptr)
+            goto done;
+
+        if (fgets(line, lineLen, file) == nullptr)
+            goto done;
+
+        // The expected format is:
+        //     $MAX $PERIOD
+        // Where "$MAX" may be the string literal "max"
+
+        space = (char*) memchr(line, ' ', lineLen);
+        if (!space)
+        {
+            assert(!"Unable to parse " CGROUP2_CPU_MAX_FILENAME " file contents.");
+            goto done;
+        }
+
+        max_quota_string = line;
+        *space = '\0';
+        period_string = space + 1;
+
+        // "max" means no cpu limit
+        if (strcmp("max", max_quota_string) == 0)
+            goto done;
+
+        errno = 0;
+        endptr = nullptr;
```

Comment:
A nit - nulling the endptr is not needed, it is an output parameter of the strtoll.

Reply:
Thanks. Fixed.